### ISSUE TITLE
PropertyPanePropertyEditor breaks webparts consuming dyndata

### ISSUE THREAD

**Issue by JakeStanger** [pnp/sp-dev-fx-property-controls#200]
#### Category
- [ ] Enhancement
- [x] Bug
- [ ] Question

#### Version

Please specify what version of the library you are using: [1.16.0]

#### Expected / Desired Behavior / Question

Editing the properties of a webpart configured to consume dynamic data should update those properties and correctly save the changes.

#### Observed Behavior

Upon refreshing the page, the web part displays the "Something went wrong" screen with the following stacktrace:

```
ERROR:
e.split is not a function

CALL STACK:
e@https://spoprod-a.akamaihd.net/files/sp-client-prod_2019-09-06.007/sp-webpart-workbench-assembly_en-us_fee33094bb6dbf7450d47040a912efe1.js:19:2480
Sa2e/m</e.prototype.setReference@https://spoprod-a.akamaihd.net/files/sp-client-prod_2019-09-06.007/sp-webpart-workbench-assembly_en-us_fee33094bb6dbf7450d47040a912efe1.js:18:165920
DCnD/z</t.prototype._deserializeDynamicData@https://spoprod-a.akamaihd.net/files/sp-client-prod_2019-09-06.007/sp-webpart-workbench-assembly_en-us_fee33094bb6dbf7450d47040a912efe1.js:21:545748
DCnD/z</t.prototype._internalDeserialize@https://spoprod-a.akamaihd.net/files/sp-client-prod_2019-09-06.007/sp-webpart-workbench-assembly_en-us_fee33094bb6dbf7450d47040a912efe1.js:21:543050
DCnD/Ee</e.prototype._initializeWebPart@https://spoprod-a.akamaihd.net/files/sp-client-prod_2019-09-06.007/sp-webpart-workbench-assembly_en-us_fee33094bb6dbf7450d47040a912efe1.js:21:602488
DCnD/Ee</e.prototype.loadWebPart/s</<@https://spoprod-a.akamaihd.net/files/sp-client-prod_2019-09-06.007/sp-webpart-workbench-assembly_en-us_fee33094bb6dbf7450d47040a912efe1.js:21:582636
```

#### Steps to Reproduce

-  Create a new web part
- Install `@pnp/spfx-property-controls`
- Add the `PropertyPanePropertyEditor` and a `PropertyPaneDynamicFieldSet` (probably with at least one `PropertyPaneDynamicField` to the property pane
- Add the web part to a page and set values for the dynamic data options (page environment will work)
- Edit the properties JSON to change the default description value, and save
- Refresh the page

---

It looks like the issue itself lies in the framework, which doesn't like dynamic data keys being updated by anything except itself. Perhaps the component could look to filter out certain keys?

**Comment by estruyf** [comment]
This will be available in version `1.19.0`.

**Comment by AJIXuMuK** [comment]
The version has been released.
